### PR TITLE
chain: Upgrade to be compatible with gaia 2.0.4

### PR DESCRIPTION
**Pull request by sorawit**
Go through `app.go`, `bandd/main.go`, and `bandcli/main.go` to make sure it does pretty much the same thing as what is https://github.com/cosmos/gaia/tree/release/v2.0.4 to ensure our code won't miss anything.

### CODE REVIEW

File: chain/app/app.go [hunk 1]
```diff
@@ -258,6 +264,7 @@ func NewBandApp(
 
 	app.mm.SetOrderBeginBlockers(mint.ModuleName, distr.ModuleName, slashing.ModuleName)
 	app.mm.SetOrderEndBlockers(zoracle.ModuleName, gov.ModuleName, staking.ModuleName)
+	app.mm.SetOrderEndBlockers(crisis.ModuleName, gov.ModuleName, staking.ModuleName)
```
Comment (taobun) [inline chain/app/app.go]: Let's merge these lines

File: chain/cmd/bandcli/main.go [hunk 2]
```diff
@@ -58,10 +59,11 @@ func main() {
 		client.NewCompletionCmd(rootCmd, true),
 	)
 
-	executor := cli.PrepareMainCmd(rootCmd, "REQ", app.DefaultCLIHome)
+	executor := cli.PrepareMainCmd(rootCmd, "BAND", app.DefaultCLIHome)
```
Comment (taobun) [inline chain/cmd/bandcli/main.go:62]: WOW

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/app/app.go [hunk 3]
```diff
@@ -31,18 +32,18 @@ import (
 )
 
 const (
-	appName          = "band"
+	appName          = "BandApp"
```
Comment (prin-r) [inline chain/app/app.go:35]: Not relevant to the "Upgrade to be compatible with gaia 2.0.4"
But it's ok.

**Review by prin-r — APPROVED** [review]

File: chain/app/app.go (hunk 3 above)
Comment (taobun) [reply chain/app/app.go]: True

**Review by taobun — APPROVED** [review]

**Pull request merged**